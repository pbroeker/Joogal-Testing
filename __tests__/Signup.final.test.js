import React from 'react';
import Final from '../Signup/Signup.Final';
import { fireEvent, render } from '@testing-library/react-native';

describe('Signup final', () => {
  const setSignup = jest.fn();
  const navigation = {navigate : jest.fn()};
  const submitSignup = jest.fn();
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = render( <Final navigation={navigation} setSignup={setSignup} submitSignup={submitSignup} /> ); 
  })

  test('snapshot', () => {
    const { toJSON } = renderedComponent;
    expect(toJSON()).toMatchSnapshot();
  })

  test('calls submitSignup correctly', () => {
    const { getByText } = renderedComponent;
    fireEvent(getByText('Continue'), 'onPress');
    expect(submitSignup).toHaveBeenCalledTimes(1);
  })
})

