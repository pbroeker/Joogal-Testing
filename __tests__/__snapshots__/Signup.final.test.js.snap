// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup final snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1d1e20",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": "20%",
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontFamily": "PoiretOne_400Regular",
          "fontSize": 30,
          "marginBottom": 30,
          "marginLeft": "10%",
        }
      }
    >
      Almost there..
    </Text>
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontFamily": "PoiretOne_400Regular",
          "fontSize": 30,
          "marginBottom": 30,
          "marginLeft": "10%",
        }
      }
    >
      Add some examples of your work in your profile
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 370,
          "marginTop": 266,
          "width": 375,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 392,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "teal",
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
      testID="finalContinueButton"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "600",
          }
        }
      >
        Continue
      </Text>
    </View>
  </View>
</View>
`;
